13_factorial/factorial.spec.js

```diff
@@ -0,0 +1,31 @@
+const factorial = require("./factorial-solution");
+
+describe('factorial', () => {
+  test('4th factorial number is 24', () => {
+    expect(factorial(4)).toBe(24);
+  });
+  test.skip('6th factorial number is 720', () => {
+    expect(factorial(6)).toBe(720);
+  });
+  test.skip('10th factorial number is 3628800', () => {
+    expect(factorial(10)).toBe(3628800);
+  });
+  test.skip('15th factorial number is 1307674368000', () => {
+    expect(factorial(15)).toBe(1307674368000);
+  });
+  test.skip('25th factorial number is 1.5511210043330986e+25', () => {
+    expect(factorial(25)).toBe(1.5511210043330986e+25);
+  });
+  test.skip('0th factorial number is 1', () => {
+    expect(factorial(0)).toBe(1);
+  });
+  test.skip('doesn\'t accept negatives', () => {
+    expect(factorial(-25)).toBe(undefined);
+  });
+  test.skip('doesn\'t accept floats', () => {
+    expect(factorial(5.4)).toBe(undefined);
+  });
+  test.skip('doesn\'t accept strings', () => {
+    expect(factorial('5')).toBe(undefined);
+  });
```

Comment:
As opposed to naming the test "doesn't accept strings", perhaps "only accepts numbers"?
And perhaps add a second test with the same description that passes in like, an array or some other non-number value?

Reply:
> As opposed to naming the test "doesn't accept strings", perhaps "only accepts numbers"? And perhaps add a second test with the same description that passes in like, an array or some other non-number value? Or a 2nd `expect` within that test block.

How about

```js
  describe('only accepts numbers', () => {
    test('doesn\'t accept negatives', () => {
      expect(factorial(-25)).toBe(undefined);
    });
    test('doesn\'t accept floats', () => {
      expect(factorial(5.4)).toBe(undefined);
    });
    test('doesn\'t accept strings', () => {
      expect(factorial('5')).toBe(undefined);
    });
    test('doesn\'t accept arrays', () => {
      expect(factorial([5])).toBe(undefined);
    });
    test('doesn\'t accept objects', () => {
      expect(factorial({})).toBe(undefined);
    });
    test('doesn\'t accept NaN', () => {
      expect(factorial(NaN)).toBe(undefined);
    });
  });
```

Reply:
I think that's a bit much - perhaps just something like rejecting `'5'`, and something like `'foo'` or `[5]` is more than sufficient for testing rejection of non-numbers.
Negatives and floats can be their own tests.

(Also `-25`, `5.4` and `NaN` are numbers, so wouldn't fit that describe block anyway).